### PR TITLE
Support for the JSON.CLEAR command

### LINKED ISSUE #1812: Add support for the JSON.CLEAR command
Please refer to https://github.com/apache/kvrocks/issues/1805.

### PR DESCRIPTION
**Pull request by 2rueSid**
support JSON.CLEAR, this closes #1812 
please review @mapleFU

[JSON.CLEAR command](https://redis.io/commands/json.clear)

### CODE REVIEW

**Comment by torwig** [comment]
@2rueSid You can format your code with the `./x.py` command. It allows your code to pass the lint stage in the CI pipeline.

File: kvrocks.conf [hunk 1]
```diff
@@ -858,4 +858,4 @@ rocksdb.write_options.memtable_insert_hint_per_batch no
 rocksdb.rate_limiter_auto_tuned yes
 
 ################################ NAMESPACE #####################################
-# namespace.test change.me
+# namespace.test change.me
```
Comment (git-hulk) [inline kvrocks.conf]: Can revert this change

File: src/common/status.h [hunk 2]
```diff
@@ -168,7 +168,7 @@ struct StringInStatusOr<T, std::enable_if_t<sizeof(T) < sizeof(std::string)>> :
   StringInStatusOr(StringInStatusOr<U>&& v) : BaseType(new std::string(*std::move(v))) {}  // NOLINT
   template <typename U, typename std::enable_if_t<!StringInStatusOr<U>::inplace, int> = 0>
   StringInStatusOr(StringInStatusOr<U>&& v)  // NOLINT
-      : BaseType((typename StringInStatusOr<U>::BaseType &&)(std::move(v))) {}
+      : BaseType((typename StringInStatusOr<U>::BaseType&&)(std::move(v))) {}
```
Comment (git-hulk) [inline src/common/status.h]: Should revert this file. It seems the different behaviors between different versions of clang-format. For clang-format >= 14, it will add a space for type and parameter name, but not before clang-format 14.

**Comment by git-hulk** [comment]
Rest are good to me, great work! @2rueSid

File: src/common/status.h (hunk 2 above)
Comment (mapleFU) [reply src/common/status.h]: We can add a clang-format rule for this? Seems check by clang-tidy version is not a good way.

**Review by mapleFU — COMMENTED** [review]
Overall LGTM except some style problems!
Thanks!

File: src/types/json.h [hunk 3]
```diff
@@ -172,6 +172,34 @@ struct JsonValue {
...
+                                         bool is_object = val.is_object() && !val.empty();
+                                         bool is_number = val.is_number() && val.as<double>() != 0;
+
+                                         if (!is_number && !is_array && !is_object) {
+                                           return;
+                                         }
+
+                                         if (is_array) val = jsoncons::json::array();
+                                         if (is_object) val = jsoncons::json::object();
+                                         if (is_number) val = 0;
```
Comment (PragmaTwice) [inline src/types/json.h]:
```suggestion
                                         if (is_array) val = jsoncons::json::array();
                                         else if (is_object) val = jsoncons::json::object();
                                         else if (is_number) val = 0;
                                         else return;
```

File: src/common/status.h (hunk 2 above)
Comment (2rueSid) [reply src/common/status.h]: Ok, I will check the versions and will revert it.

Comment (2rueSid) [reply src/common/status.h]: I reverted the changes. It was because I use clang-format version 17.*.

Should I investigate this a bit more and try to add a rule?


Also, I'm trying to run linter with the command (actually each time I try to run linter it fails):
```bash
$ ./x.py check tidy --clang-tidy-path /.../llvm@14/bin/clang-tidy --run-clang-tidy-path /.../llvm@14/bin/run-clang-tidy
```

and receive an output:

```
4975 warnings and 1 error generated.
Error while processing kvrocks/src/types/bloom_filter.cc.
Suppressed 4974 warnings (4966 in non-user code, 8 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
9 warnings treated as errors
Traceback (most recent call last):
  File "./x.py", line 422, in <module>
    args.func(**arg_dict)
  File "./x.py", line 195, in clang_tidy
    run(run_command, *options, *regexes, verbose=True, cwd=basedir)
  File "./x.py", line 68, in run
    raise RuntimeError(err)
RuntimeError:
failed to run: ('/opt/homebrew/opt/llvm@14/bin/run-clang-tidy', '-p', 'build', '-clang-tidy-binary', '/opt/homebrew/opt/llvm@14/bin/clang-tidy', '-j8', '-header-filter=kvrocks/src/|utils/kvrocks2redis/|tests/cppunit/', 'kvrocks/src/', 'utils/kvrocks2redis/', 'tests/cppunit/')
exit with code: 1
```

And the problem is, I guess that it's not a normal behavior, and I can't see an actual error, or if it actually exists. Are there some suggestions which I can try to resolve this?

Comment (mapleFU) [reply src/common/status.h]: Let's do it in a separate patch, since it's not a good idea making this patch too complex

Comment (2rueSid) [reply src/common/status.h]: Sure, you can assign me to an issue and I will do it ASAP.

**Review by torwig — APPROVED** [review]
Generally, LGTM.

**Review by mapleFU — APPROVED** [review]
Will wait for twice's review

**Review by git-hulk — APPROVED** [review]

**Review by PragmaTwice — APPROVED** [review]

**Comment by PragmaTwice** [comment]
Thanks for your contribution!
